Makefile

```diff
@@ -306,6 +313,29 @@ $(TARGET_BIN): $(TARGET_ELF)
 	@echo "Creating BIN $(TARGET_BIN)" "$(STDOUT)"
 	$(V1) $(OBJCOPY) -O binary $< $@
 
+$(TARGET_S19): $(TARGET_ELF)
+	@echo "Creating srec/S19 $(TARGET_S19)" "$(STDOUT)"
+	$(V1) $(OBJCOPY) --output-target=srec $(TARGET_S19)
+
+$(TARGET_EXST): $(TARGET_BIN)
+	@echo "Creating EXST $(TARGET_EXST)" "$(STDOUT)"
+# Linker script should allow .bin generation from a .elf which results in a file that is the same length as the FIRMWARE_SIZE.
+# These 'dd' commands will pad a short binary to length FIRMWARE_SIZE.
+	$(V1) dd if=/dev/zero ibs=1k count=$(FIRMWARE_SIZE) of=$(TARGET_EXST)
+	$(V1) dd if=$(TARGET_BIN) of=$(TARGET_EXST) conv=notrunc
+
+	@echo "Generating MD5 hash of binary" "$(STDOUT)"
+	$(V1) openssl dgst -md5 $(TARGET_EXST) > $(TARGET_BIN).md5 
```

Comment:
The specification for the EXST format looks mostly sound. The requirement for the reserved bytes to be all 0xaa seems to be a bit unreasonable - this will make it impossible to ever use these bytes without breaking compatibility with any boot loader that expects the specification to be met - this should be removed.

Can I also propose that we allocate the first of the reserved bytes (or the last one, as it is closest to the checksum) as an indicator of what checksum type is used - this will pave the way to use a less problematic algorithm when moving to Universal Targets.

Reply:
updated documentation and linker script.  as noted using the one closest to the checksum isn't possible as checksum lengths are all different.  documentation updated to include a block format.